example Milton Keynes network GDF as utility func

jGaboardi opened the pull request:
* This PR resolves #208
* code block in `README` confirmed functional 
* *adaptive simplification* notebook rerun for verification
* doctest in `artifacts.FaceArtifacts` passes locally
* docs rebuild locally without issue

martinfleis commented on README.md:
```diff
@@ -11,7 +11,10 @@
 import neatnet
 
-simplified = neatnet.neatify(gdf)
+# Milton Keynes, UK
+original = neatnet.utils.fetch_milton_keynes()
+
+simplified = neatnet.neatify(original)
```
I hate being a pain, but I am not sure I like this. It is compely unclear without running the code what is `original`. While `gdf` indicates a geodataframe. Yeah, it runs now but not sure it sends the message as it should.

martinfleis commented on docs/source/intro.ipynb:
Can we keep the explicit code using osmnx here? So people know how to fetch and process their own data?

martinfleis commented on neatnet/artifacts.py:
```diff
@@ -68,16 +68,20 @@ class FaceArtifacts:
     --------
-    >>> fa = neatnet.FaceArtifacts(street_network_prague)
+    >>> import neatnet
+    >>> street_network = neatnet.utils.fetch_milton_keynes()
```
Can you print head of `street_network` after this line before using neatnet? It is unclear what this is. A graph? A gdf? Path to a file? Something else?

martinfleis commented on pyproject.toml:
```diff
@@ -32,6 +32,7 @@ dependencies = [
     "momepy>=0.9.0",
     "networkx>=3.3",
     "numpy>=2",
+    "osmnx>=2",
```
Can we make this optional?

martinfleis reviewed (commented):
Runnable examples are great but we should make effort to make this more transparent. The custom function obfuscates what is actually going on and how to adapt code to users' own data. I am fine with this, pending my other comments, but sometimes it is easier to drop reproducibility in sake of simplicity.

jGaboardi commented:
> Runnable examples are great but we should make effort to make this more transparent. The custom function obfuscates what is actually going on and how to adapt code to users' own data.

I agree. The whole reason for this is so we don't have to copy & paste the Milton Keynes code block wherever we are doing an example.

How about this compromise:

* We simply drop the `Example` docstring section in `FaceArtifacts`
* Then there is no real reason to have the baked-in helper
* And we can have that full explicit Milton Keynes block in the `README` and `intro.ipynb`
* Then is makes perfect sense to drop the `osmnx` hard dep in `pyproject.toml`

> I hate being a pain

highly doubt that -- it is your hobby! 😆 -- Joking of course. Your comments/critiques are always welcome and valuable for a better outcome.

martinfleis commented:
One way of doing this more transparently is to store the GPKG somewhere, expose it within geodatasets and load it as a one-liner with geopandas.

jGaboardi commented:
> One way of doing this more transparently is to store the GPKG somewhere, expose it within geodatasets and load it as a one-liner with geopandas.

I like this idea. We could actually plop the [Apalachicola testing data](https://github.com/uscuni/neatnet/tree/main/neatnet/tests/data) into said GeoPackage and keep all sample data tidy in one place for a low overhead. If this sounds amenable, would that perhaps be stored as something like `neatnet_sample_data` or do we think it better to keep the Milton Keynes and Apalachicola stuff separate?

jGaboardi commented:
@martinfleis 

> I like this idea. We could actually plop the [Apalachicola testing data](https://github.com/uscuni/neatnet/tree/main/neatnet/tests/data) into said GeoPackage and keep all sample data tidy in one place for a low overhead. If this sounds amenable, would that perhaps be stored as something like neatnet_sample_data or do we think it better to keep the Milton Keynes and Apalachicola stuff separate?

Should we revisit this idea now that [#214](#214) is tackled?

Some key points ideas:
* perhaps 1 single GPKG with layers for:
  *  Apalachicola testing data (original, simplified result, simplified result with exclusion mask) -- maybe also adding [the exclusion mask](https://github.com/uscuni/neatnet/blob/d20e7dfa903a63bd8182621c0c04c442c67bdc2c/neatnet/tests/test_simplify.py#L20-L42) itself in there
  * Milton Keynes stuff
  * anything else???
* what about having our FUA stuff from `data/` in there?

jGaboardi commented:
If any of the above sounds good, then probably I'll open a fresh issue to continue this discussion.

jGaboardi commented:
> Still haven't made my mind on how to approach the data storage...

@martinfleis @anastassiavybornova – any further thoughts on data storage?
